OpenRA.Mods.Common/Traits/AmmoPool.cs

```diff
@@ -105,23 +101,40 @@ public void Attacking(Actor self, Target target, Armament a, Barrel barrel)
 
 		public void Tick(Actor self)
 		{
-			if (!Info.SelfReloads)
+			if (reload == null || reload.IsTraitDisabled)
 				return;
 
 			// Resets the tick counter if ammo was fired.
-			if (Info.ResetOnFire && GetAmmoCount() < PreviousAmmo)
+			if (reload.Info.ResetOnFire && GetAmmoCount() < PreviousAmmo)
 			{
-				RemainingTicks = Info.SelfReloadTicks;
+				RemainingTicks = reload.Info.ReloadTicks;
 				PreviousAmmo = GetAmmoCount();
 			}
 
 			if (!FullAmmo() && --RemainingTicks == 0)
```

Comment:
Yeah it seems to decrement RemainingTicks even if !FullAmmo() is false, so if !FullAmmo stays false for a while then RemainingTicks eventually goes below 0.